### ISSUE TITLE
Write dev environment setup guide

### ISSUE THREAD

**Issue by Thing342** [TravelMapping/Web#90]
Currently, there is no guide to setting up a development environment for the project, meaning that setting one up is a pain and subject to a large amount of trial and error. I think it would be helpful to new developers to write a quickstart guide (either on the wiki or on the forums) on how to set up a new dev environment so that we can get new devs up and running fairly quickly.

**Comment by jteresco** [comment]
Yes, definitely.  I'm also open to ways to change the system so it more easily supports the idea of development, staging, and production servers.

**Comment by Thing342** [comment]
I'll get started on writing a draft (focused on unix-like systems, since that's what I use) tonight. Someone with more experience developing on Windows can handle that portion.

**Comment by ssoworld** [comment]
What specifically are we targeting?


We should focus on set-up of Apache, PHP, Python and MySQL versions.  I tried setting up an environment on my local linux box and I believe I am suffering from "version-itis", though I got most of it working.


________________________________

**Comment by Thing342** [comment]
I'm thinking the entire development stack (PHP, MySQL, and Python comonents), so that the reader can follow the guide and then immediately be able to contribute to the project with no further setup.
---
Scott, here are the proper software packages (AFAIK):
php5-mysqlnd (PHP 5.6 w/ mysqli)
MySQL Community Server, latest version (5.7 last I checked)
Python 3
Not sure about Apache (I just use the dev server built into PHP for debugging)

**Comment by jteresco** [comment]
That list looks reasonable.  We're also using Apache 2.4.17 but I don't think there's anything specific to that version that we're doing.  It should work in pretty much any web server.  Python is specifically now 3.5.1 but I don't think I'm doing anything that wouldn't work in other fairly recent Python releases.  I'm only on PHP 5.4.45.

One item that is specific to Apache is the way I deny access to the new config file that will contain the information about the DB name and credentials and the Google Maps API key.  I hide that file with an entry in my httpd.conf file.  Development servers will be less concerned about this anyway, I'd think.

**Comment by jteresco** [comment]
We're part way there on this with new information in README files in the repositories.

**Comment by yakra** [comment]
We've had the new instructions on how to use datacheck.py.
I'd love it if instructions on how to set up my own server were made equally easy.

**Comment by jteresco** [comment]
Do we know of anyone other than me and @yakra who have set up development servers?  Is there information we should add to the https://github.com/TravelMapping/Web/blob/master/README.md file, or is there a need for more detailed instructions?

**Comment by yakra** [comment]
>Do we know of anyone other than me and @yakra who have set up development servers?

I think *maybe* @Thing342 may have set up a server back in the day?

>  Is there information we should add to the https://github.com/TravelMapping/Web/blob/master/README.md file, or is there a need for more detailed instructions?

There were some minor hiccups; it's been long enough that I can't really remember what. I was too busy barrelling on through puzzling out the process & learning as I went to stop & note what might need to be clarified.
* I think I left a GitHub issue somewhere about... something about the MySQL setup, users, permissions?...
* Not sure if the info about tm.conf & the symbolic links is up to date. (We create two files, right? And localupdate.sh sets the links to them?)

There are a few issues specific to CentOS, most notably setting SELinux context for files served by Apache. ...But how deep do we want to get into the weeds on this?
